docs/source/1.0/spec/aws/aws-core.rst

```diff
@@ -1219,6 +1219,301 @@ Clients SHOULD use an LRU cache implementation with an initial cache limit of
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - requestAlgorithmMember
+      - ``string``
+      - Defines a top-level operation input member that is used to configure
+        request checksum behavior. The input member MUST target a string shape
+        marked with the :ref:`enum-trait`. Each value in the enum represents a
+        supported checksum algorithm. Algorithms must be one of the following
+        supported values: "crc32c", "crc32c", "sha1", or "sha256".
+    * - requestChecksumRequired
+      - ``boolean``
+      - Indicates an operation requires a checksum in its HTTP request. By
+        default, the checksum used for a service is an MD5 checksum passed
+        in the Content-MD5 header. When the input member represented by the
+        ``requestAlgorithmMember`` property is set, the default behavior is
+        disabled.
+    * - requestValidationModeMember
+      - ``string``
+      - Defines a top-level operation input member used to opt-in to
+        best-effort validation of a checksum returned in the HTTP response of
+        the operation. The input member MUST target a string shape marked with
+        the :ref:`enum-trait` that contains the value "ENABLED".
+    * - responseAlgorithms
+      - ``set<string>``
+      - Defines the checksum algorithms clients SHOULD look for when validating
+        checksums returned in the HTTP response. Each algorithm must be one of
+        the following supported values: "crc32c", "crc32", "sha1", or "sha256".
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        @httpChecksum(
+            requestChecksumRequired: true,
+            requestAlgorithmMember: "checksumAlgorithm",
+            requestValidationModeMember: "validationMode",
+            responseAlgorithms: ["crc32c", "crc32", "sha1", "sha256"]
+        )
+        operation PutSomething {
+            input: PutSomethingInput,
+            output: PutSomethingOutput
+        }
+
+        structure PutSomethingInput {
+            @httpHeader("x-amz-request-algorithm")
+            checksumAlgorithm: ChecksumAlgorithm,
+
+            @httpHeader("x-amz-response-validation-mode")
+            validationMode: ValidationMode,
+
+            @httpPayload
+            content: Blob,
+        }
+
+        @enum([
+            {
+                value: "crc32c",
+                name: "CRC32C"
+            },
+            {
+                value: "crc32",
+                name: "CRC32"
+            },
+            {
+                value: "sha1",
+                name: "SHA1"
+            },
+            {
+                value: "sha256",
+                name: "SHA256"
+            }
+        ])
+        string ChecksumAlgorithm
+
+        @enum([
+            {
+                value: "ENABLED",
+                name: "ENABLED"
+            }
+        ])
+        string ValidationMode
+
```

Comment:
Can you show an example of an invalid trait that doesn't define request or response settings so it fails validation?